gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/DagManagementDagActionStoreChangeMonitor.java

```diff
@@ -37,16 +40,56 @@
 @Slf4j
 public class DagManagementDagActionStoreChangeMonitor extends DagActionStoreChangeMonitor {
   private final DagManagement dagManagement;
+  @VisibleForTesting @Getter
+  private final DagActionReminderScheduler dagActionReminderScheduler;
 
   // Note that the topic is an empty string (rather than null to avoid NPE) because this monitor relies on the consumer
   // client itself to determine all Kafka related information dynamically rather than through the config.
   public DagManagementDagActionStoreChangeMonitor(Config config, int numThreads,
       FlowCatalog flowCatalog, Orchestrator orchestrator, DagManagementStateStore dagManagementStateStore,
-      boolean isMultiActiveSchedulerEnabled, DagManagement dagManagement) {
+      boolean isMultiActiveSchedulerEnabled, DagManagement dagManagement, DagActionReminderScheduler dagActionReminderScheduler) {
```

Comment:
do u need to make sure initialization order in guide works here?

Reply:
DagActionReminderScheduler only needs StdSchedulerFactory, which does not need any other dependency for initialization, so there should not be any circular dependency.
If creation of DagActionReminderScheduler were indirectly dependent on DagManagementDagActionStoreChangeMonitorFactory, that would have been a problem.